CHANGES

```diff
@@ -1,3 +1,7 @@
+20190710 code: add instchown to set ownership and mode on installed files.
+20190709 code: add instpackage to install without root access.
+20190714 code: removed idedit, install-big.  var-qmail packages are no more.
```

Comment:
Sort by date?

Reply:
No, but the reason bears explaining.  There are two dates here, one explicit and one implicit.  The explicit date is when the change was made, the implicit date is the order of the lines in the file, which is the order the commits went in to master.

This file is going to create merge conflicts on rebase, frequently (It's the nature of ChangeLog files and there are format changes we could make that then provide tooling to make it easier to manage, but that's a standalone PR and discussion), because the CHANGES file changes with every substantial commit.

To make it easier to merge, as well as coherent to follow by reading in order, I've kept these lines in commit order, even when the date the change was made is thereby out of order.  By doing this, every change that modifies CHANGES places it's entry at the top of that file, and resolving conflicts is a matter of putting them in commit order, usually by taking non-context lines from both changes in the order presented in a merge conflict.  (Only possible if commit order of CHANGES is maintained.)

To get the implicit date (commit order) and the explicit date (as stated in CHANGES) to match we'd need to invent or triangulate on a date that is in-order for both cases--which is naturally the date we rebase on to master.

Enforcing that will likely (as it would here) result in a last-minute entry to CHANGES across each commit in the series, just to "falsify" the explicit date by moving it forward in time so the whole series has a single date, right before we rebase on to master.  Possible.  Burdensome.

This will result in more merge conflicts than we already create by maintaining a ChangeLog.  Instead of doing that we should use tooling and file formats designed to solve this problem, and leave these lines in-order qua commit history but out-of-order qua date-of-commit until and unless we've done so.  (A quick read of pages return from the web search "changelog merge conflict" will give you a reasonable overview of the problem.)